src/inmanta/agent/agent.py

```diff
@@ -843,11 +844,40 @@ async def setup_executor(
             executor = None
             failed_resource_types = resource_types
 
+        return executor, invalid_resource_types, failed_resource_types
+
+    async def load_configuration_model(
+        self,
+        model_version: int,
+        action: const.ResourceAction,
+        resource_batch: list[JsonType],
+        resource_types: set[ResourceType],
+    ) -> tuple[dict[ResourceVersionIdStr, const.ResourceState], list[ResourceDetails], Optional[executor.Executor]]:
+        """
+        Load (part of) a configuration model. This method is expected to be called as an
+        initialization step when interacting with resources.
```

Comment:
We don't really load the model, do we? Just the resources? And we don't even really load them, just request an executor and "parse" them (for lack of a better word). I fail to come up with satisfying name though.Things that come to mind are `get_executable_resources`, `get_executor_resources`, `prepare_resources`, `setup_for_execution`.

Reply:
I think I like `prepare_resources` the most, since it will be used in `dryrun` and `getfacts` contexts as well. I always associate (maybe wrongfully so?) the `execute` semantic with the `deploy` itself